src/pages/DatasetCatalog.js

```diff
@@ -251,24 +250,35 @@ export default function DatasetCatalog(props) {
     return true;
   };
 
+  const filteredAndPaginatedDatasetList = () => {
+    return datasetList.filter(searchTable(searchDulText)).slice((currentPage - 1) * pageSize, currentPage * pageSize);
+  };
+
+  const allOnPageSelected = () => {
+    return filteredAndPaginatedDatasetList().every((row) => {
+      return row.checked || (!row.active);
```

Comment:
The Select Visible checkbox appears checked when the datasets are still loading since this condition is technically true. You can also make sure that the dataset list is not empty here to fix it